src/software/thunderscope/robot_diagnostics/drive_and_dribbler_widget.py

```diff
@@ -1,28 +1,22 @@
 from pyqtgraph.Qt.QtCore import Qt
 from pyqtgraph.Qt.QtWidgets import *
 import time
+import software.python_bindings as tbots
 
 from software.thunderscope.thread_safe_buffer import ThreadSafeBuffer
 from software.thunderscope.common import common_widgets
 from proto.import_all_protos import *
 
-# TODO (#2683) get these values from robot constants
-MAX_DRIBBLER_RPM = 10000
-MIN_DRIBBLER_RPM = -10000
 
-MAX_LINEAR_SPEED_MPS = 5
-MIN_LINEAR_SPEED_MPS = -5
-
-MAX_ANGULAR_SPEED_RAD_PER_S = 20
-MIN_ANGULAR_SPEED_RAD_PER_S = -20
+class DriveAndDribblerWidget(QWidget):
+    def __init__(self, proto_unix_io):
+        self.constants = tbots.create2021RobotConstants()
 
 
 class DriveAndDribblerWidget(QWidget):
     def __init__(self, proto_unix_io):
         """Initialize the widget to control the robot's motors
-
         :param proto_unix_io: the proto_unix_io object
-
         """
         self.input_a = time.time()
```

Comment:
You don't need to declare the `DriveAndDribblerWidget` class again. You can just define `self.constants` inside the existing class.